### DISCUSSION TITLE
Database Maintenance

### DISCUSSION THREAD

**Discussion by FaFre** [simolus3/drift#2431] [Q&A]
Having a long living database, from time to time there is some maintenance required. For example running `ANALYZE`, maybe `VACUUM` every couple of weeks, or optimizing the FTS5 index. 

Those operations can be time intensive and are blocking the database. So it shouldn't be done while the user is interacting with the app.

Where would be a good place to perform those tasks?

**Comment by simolus3** [comment]
> Those operations can be time intensive and are blocking the database. So it shouldn't be done while the user is interacting with the app.

I think this depends on how sophisticated you want to make this / how long these operations actually take. If it's less than a second, perhaps you could run them at the start of your app (e.g. while you're showing a splash/loading screen)?

If they take much longer, you could setup a periodic background task with platform plugins that just does nothing if the app happens to be running at the same time.
